agent/src/com/jonnyzzz/teamcity/virtual/run/docker/DockerVM.java

```diff
@@ -125,12 +139,14 @@ public BuildProcess startImpl() throws RunBuildException {
       }
 
       @NotNull
-      private List<String> scriptRun(@NotNull final File script) {
-        return Arrays.asList(
-                ctx.getShellLocation(),
-                "-c",
-                "\"source " + script.getName() + "\""
-        );
+      private List<String> scriptRun(@NotNull final File script) throws RunBuildException {
+        if(ctx.isDockerServerWindowsBased() && ctx.getShellLocationInsideContainer().equals("cmd.exe")) {
+          return  Arrays.asList(ctx.getShellLocationInsideContainer(), "/C", "call " + script.getName());
+        } else if(ctx.isDockerServerWindowsBased() && ctx.getShellLocationInsideContainer().equals("powershell.exe")) {
+          return  Arrays.asList(ctx.getShellLocationInsideContainer(), "-Command", "get-content " + script.getName() + " | Invoke-Expression");
```

Comment:
Yes, I agree. User can call powershell.exe from cmd like he can call powershell from bash. Powershell is cross platform for a while already.